x-pack/plugin/sql/qa/server/src/main/java/org/elasticsearch/xpack/sql/qa/rest/RestSqlTestCase.java

```diff
@@ -1433,6 +1436,19 @@ public void testAsyncTextPaginated() throws IOException, InterruptedException {
         }
     }
 
+    public void testCompressCursor() throws IOException {
+        String doc = IntStream.range(0, 1000)
+            .mapToObj(i -> String.format(Locale.ROOT, "\"field%d\": %d", i, i))
+            .collect(Collectors.joining(","));
+        index("{" + doc + "}");
+
+        String mode = randomMode();
+        Map<String, Object> resp = toMap(runSql(query("SHOW COLUMNS FROM " + indexPattern("test")).fetchSize(1).mode(mode)), mode);
+
+        // without compression, the cursor is at least <avg. fieldname length> * 1000 bytes (in fact it is ~35kb)
+        assertThat(resp.get("cursor").toString().length(), lessThan(5000));
```

Comment:
Would a decompression check (i.e. no exception thrown) or some `COMPRESSOR.isCompressed()` make sense here?

Reply:
I also think that would be helpful but unfortunately not too trivial to implement because the wire format is now a bit more intricate than just a compressed byte array encoded in base64.